src/main/java/tigase/xmpp/impl/BindResource.java

```diff
@@ -74,15 +83,21 @@ public String id() {
 		return ID;
 	}
 
-	public void setResourcePrefix(String resourcePrefix) {
-		this.resourcePrefix = resourcePrefix;
-		this.resourceDefPrefix =
-				Math.abs(DNSResolverFactory.getInstance().getDefaultHost().hashCode()) + "-" + (this.resourcePrefix != null ? this.resourcePrefix : resourceDefPrefix);
-	}
-
 	@Override
 	public boolean preProcess(Packet packet, XMPPResourceConnection session, NonAuthUserRepository repo,
 							  Queue<Packet> results, Map<String, Object> settings) {
+		if (packet.getServerAuthorisedStanzaFrom().isPresent()) {
+			// Correct user JID is already available in the packet so let's use it and short-circuit further processing.
+			packet.initVars(packet.getServerAuthorisedStanzaFrom().get(), packet.getStanzaTo());
+			return false;
```

Comment:
I'm not sure if we should short-circuit this here. I know "task" is done, but if some new code would be added to this method in "typical processing place" it could not catch those packets leaving here..

Reply:
Could you expand on this? What other processing would you have in mind? It seems that the intention of the pre-processor is to simply stamp packets and actual Bind processing is done in process method.

Alternatively we could have "stamper" as separate pre-processor.

Reply:
Yes, the idea is to just stamp packets, but it may evolve with time.. ie. we could add some other checks in here